src/net_processing.cpp

```diff
@@ -1709,7 +1712,7 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
             } else {
                 LogPrint("mempool", "not keeping orphan with rejected parents %s\n",tx.GetHash().ToString());
             }
-        } else {
+        } else { // not accepted and not missing inputs, i.e. invalid
```

Comment:
That is the intended and advertised purpose of the whitelisting, read the comment in the else block!